### PR TITLE
Move water residue names to residue_names.py

### PR DESCRIPTION
**Pull request by kyleabeauchamp**
See #620

### CODE REVIEW

File: mdtraj/core/residue_names.py [hunk 1]
```diff
@@ -49,3 +49,6 @@
     'PHE', 'PRO', 'SER', 'THR', 'TRP', 'TYR', 'UNK', 'VAL'])
 
 
+_WATER_RESIDUES = frozenset([
+'H2O', 'HHO', 'OHH', 'HOH', 'OH2', 'SOL',
+                             'WAT', 'TIP', 'TIP2', 'TIP3', 'TIP4'])
```
Comment (rmcgibbo) [inline mdtraj/core/residue_names.py]: weird tabs or something?

**Comment by kyleabeauchamp** [comment]
fixed

**Comment by mpharrigan** [comment]
This is strongly related to #637 

Specifically, If we normalize our inputs properly, there should only be *one* water residue name

**Comment by cxhernandez** [comment]
Should 'HOH' still be in `_SOLVENT_TYPES ` then?

**Pull request merged**